src/Core/Checkout/Order/SalesChannel/SetPaymentOrderRoute.php

```diff
@@ -137,11 +144,12 @@ function () use ($payload, $context): void {
         $this->eventDispatcher->dispatch($event);
     }
 
-    private function validateRequest(SalesChannelContext $salesChannelContext, string $paymentMethodId): void
+    private function validateRequest(SalesChannelContext $salesChannelContext, string $paymentMethodId, OrderEntity $order): void
     {
-        $availablePayments = $this->paymentRoute->load(new Request(), $salesChannelContext, new Criteria());
+        $cart = $this->orderConverter->convertToCart($order, $salesChannelContext->getContext());
+        $response = $this->checkoutGatewayRoute->load(new Request(), $cart, $salesChannelContext);
```

Comment:
Why not use the request from the route call?

Reply:
No particular reason. Something like this? Request as new parameter and remove the paymentMethodId
```suggestion
private function validateRequest(SalesChannelContext $salesChannelContext, Request $request, OrderEntity $order): void
    {
        $paymentMethodId = $request->request->getAlnum('paymentMethodId');
        $cart = $this->orderConverter->convertToCart($order, $salesChannelContext->getContext());
        $response = $this->checkoutGatewayRoute->load($request, $cart, $salesChannelContext);

```